libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/SfapApiGenerations.kt

```diff
@@ -0,0 +1,79 @@
+package com.salesforce.androidsdk.auth
+
+import com.salesforce.androidsdk.auth.SfapApiGenerationsResponseBody.Companion.fromJson
+import com.salesforce.androidsdk.rest.RestClient
+import com.salesforce.androidsdk.rest.RestRequest
+import com.salesforce.androidsdk.rest.RestRequest.RestMethod.POST
+import kotlinx.serialization.json.Json.Default.encodeToString
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.io.IOException
+
+/**
+ * Provides generated text from the `sfap_api` "generations" endpoint.
+ *
+ * See https://developer.salesforce.com/docs/einstein/genai/guide/access-models-api-with-rest.html
+ *
+ * @param apiHostName The Salesforce `sfap_api` hostname
+ * @param modelName The model name to request generations from.  For possible
+ * values, see https://developer.salesforce.com/docs/einstein/genai/guide/api-names.html
+ * @param restClient The REST client to use
+ */
+class SfapApiGenerations(
+    private val apiHostName: String = API_HOST_NAME_PROD,
+    private val modelName: String,
+    private val restClient: RestClient
+) {
+
+    /**
+     * Fetches generated text from the `sfap_api` "generations" endpoint.
+     * @param prompt The prompt request parameter value
+     * @return The endpoint's response
+     */
+    @Throws(
+        IOException::class,
+        SfapApiException::class
+    )
+    fun fetch(
+        prompt: String
+    ): SfapApiGenerationsResponseBody {
+
+        // Request the generated text.
+        val restRequest = RestRequest(
+            POST,
+            "https://$apiHostName/einstein/platform/v1/models/$modelName/generations",
```

Comment:
@wmathurin - How's this look for building the request? Do we have any logic anywhere that already encapsulates the `api.salesforce.com` host along with its "dev" and "prod" concepts? Is there any thought that this code should accept an arbitrary sfap_api host?

Reply:
Maybe it's one of the URLs returned in the [identity service response](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/dev/libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java#L573)

But maybe not - not seeing in the [doc](https://help.salesforce.com/s/articleView?id=sf.remoteaccess_using_openid.htm&type=5)

Ideally we should get it from the server during login. Short of that, we could ask the app to pass it. We shouldn't hardcode it in the SDK.

Reply:
How about https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2644/commits/071abae988069daf93d547c7719171aaa186a7c4 so we don't have that value and the app provides it?